### PR TITLE
Feature/nested stack addon

### PR DESCRIPTION
**Pull request by shapirov103**
*Issue #, if available:* #191

*Description of changes:*
Nested stack add-on implementation and test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by shapirov103** [comment]
@kcoleman731 the nested stack for ES creation if wrapped as nested stack could use this add-on (we need to add an usage tracking id to it)

File: lib/stacks/eks-blueprint-stack.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { StackProps } from '@aws-cdk/core';
 import { IVpc, Vpc } from '@aws-cdk/aws-ec2';
 import { KubernetesVersion } from '@aws-cdk/aws-eks';
 import { MngClusterProvider } from '../cluster-providers/mng-cluster-provider';
-import { ClusterAddOn, Team, ClusterProvider, ClusterPostDeploy, AsyncStackBuilder } from '../spi';
+import { ClusterAddOn, Team, ClusterProvider, ClusterPostDeploy, AsyncStackBuilder, ClusterInfo } from '../spi';
```
Comment (askulkarni2) [inline lib/stacks/eks-blueprint-stack.ts:8]: Consider using `import * as ....` as per our new convention :)

File: lib/addons/nested-stack/index.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * Properties for the nested stack add-on.
+ */
+export class NestedStackAddOnProps {
+    /**
+     * Required identified, must be unique within the parent stack scope.
+     */
+    id: string;
+
+    /**
+     * Builder that genertes the stack.
```
Comment (askulkarni2) [inline lib/addons/nested-stack/index.ts:15]: minor: genertes -> generates

File: lib/addons/nested-stack/index.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { Construct, NestedStackProps } from "@aws-cdk/core";
+import { IdentityStore } from "aws-sdk";
```
Comment (askulkarni2) [inline lib/addons/nested-stack/index.ts:2]: Seems like an odd import here.

**Review by askulkarni2 — APPROVED** [review]
Minor comments.

**Review by kcoleman731 — DISMISSED** [review]
Approved pending @askulkarni2 comments

File: lib/stacks/eks-blueprint-stack.ts (hunk 1 above)
Comment (shapirov103) [reply lib/stacks/eks-blueprint-stack.ts]: Addressed

File: lib/addons/nested-stack/index.ts (hunk 2 above)
Comment (shapirov103) [reply lib/addons/nested-stack/index.ts]: Addressed

File: lib/addons/nested-stack/index.ts (hunk 3 above)
Comment (shapirov103) [reply lib/addons/nested-stack/index.ts]: Fixed

**Pull request merged**